### DISCUSSION TITLE
v7.4.0 Release of PowerShell

### DISCUSSION THREAD

**Discussion by adityapatwardhan** [PowerShell/PowerShell#20700] [General]
### General Cmdlet Updates and Fixes

- Added a missing `ConfigureAwait(false)` call to webcmdlets so they don't block (#20622)
- Fix `Group-Object` so output uses current culture (#20623)
- Block getting help from network locations in restricted remoting sessions (#20615)

### Build and Packaging Improvements

<details>

<summary>

<p>Bump .NET 8 to 8.0.0 RTM build</p>

</summary>

<ul>
<li>Add internal .NET SDK URL parameter to release pipeline (Internal 28474)</li>
<li>Update the CGManifest file for v7.4.0 release (Internal 28457)</li>
<li>Fix repository root for the nuget.config (Internal 28456)</li>
<li>Add internal nuget feed to compliance build (Internal 28449)</li>
<li>Copy azure blob with PowerShell global tool to private blob and move to CDN during release (Internal 28438)</li>
<li>Fix release build by making the internal SDK parameter optional (#20658) (Internal 28440)</li>
<li>Make internal .NET SDK URL as a parameter for release builld (#20655) (Internal 28428)</li>
<li>Update <code>PSResourceGet</code> version for <code>1.0.1</code> release (#20652) (Internal 28427)</li>
<li>Bump .NET 8 to 8.0.0 RTM build (Internal 28360)</li>
<li>Remove Auth header content from ErrorRecord (Internal 28409)</li>
<li>Fix setting of variable to consume internal SDK source (Internal 28354)</li>
<li>Bump Microsoft.Management.Infrastructure to v3.0.0 (Internal 28352)</li>
<li>Bump <code>Microsoft.PowerShell.Native</code> to <code>v7.4.0</code> (#20617) (#20624)</li>
</ul>

</details>

[7.4.0]: https://github.com/PowerShell/PowerShell/compare/v7.4.0-rc.1...v7.4.0

### SHA256 Hashes of the release artifacts

- hashes.sha256
  - 891C46EA85EF10695C1CDD3ED6A5FD3DC670CAE79C21ACA2CE6EDD99F75CE104
- powershell_7.4.0-1.deb_amd64.deb
  - 550331E243A7311E1BF0CA83D8C96C5E78BA82A765C4A9BA34AA0DB6EB9A9F22
- powershell-7.4.0-1.cm.aarch64.rpm
  - BBD1E1A07B760C7A7E0097601F9D359A1E0807EC25005AFC281E4D33A3F1A19E
- powershell-7.4.0-1.cm.x86_64.rpm
  - B2FEC6676A01D2BBCEF70703BB8DA3F9E68874FF6392504EF25C32D2F0C767C1
- powershell-7.4.0-1.rh.x86_64.rpm
  - EA2282108E6FBA6B4CC9CD56E4C23BF2663A98D24BE49E5E8FBE3C1ED2E1E9A8
- powershell-7.4.0-linux-arm32.tar.gz
  - 6A5B31EF2A4976AA20E5E39A47ACAB0575F189C8D66895C0E341D6E2A6442681
- powershell-7.4.0-linux-arm64.tar.gz
  - 13D317610BA133C002990C40622A886989368460451809861F378C25730989F4
- powershell-7.4.0-linux-musl-x64.tar.gz
  - A7EAFBE19F65226C20198874991EDB5793738547E06CDEAE6654B86F189B9FC1
- powershell-7.4.0-linux-x64.tar.gz
  - AA82C3E9241503B52DB1827E3FD7FC099DA74F828BA032E9B726E50C72B4BEE5
- powershell-7.4.0-linux-x64-fxdependent.tar.gz
  - F3ABB6F8CD1EFA2ECBE0E319D24C288AC4F6653F9FA5145516400541A751DC42
- powershell-7.4.0-linux-x64-musl-noopt-fxdependent.tar.gz
  - 0CEC7E9A7C94CA7F970BE3161EC482254D75CF75ABC314D7BF224617D52A019F
- powershell-7.4.0-osx-arm64.pkg
  - AA7F452A96AAF5A1260E50ED5888BD564BB2A5B1B0FDA730542EC7831A730612
- powershell-7.4.0-osx-arm64.tar.gz
  - 1674FDD5718CAFB804EFF7585FE8B7041362322AA944E0D2F816CA1D75E6351F
- powershell-7.4.0-osx-x64.pkg
  - 08ABEDD85E9F3ED3A8C7ACF43BE6048DDC761D2C75C19F7BFDB7BE7539E9F8C4
- powershell-7.4.0-osx-x64.tar.gz
  - 8B6BD0B4DA7AC2053683ED298B396F8929FF5CA3B803065C736F53209E815C13
- PowerShell-7.4.0-win-arm64.zip
  - A47801A3FBB7E9343C2803E775098C2361E0FB134FA4A3FACB8D81A811FA9873
- PowerShell-7.4.0-win-fxdependent.zip
  - 3EFAE9EA6224E656C54A3BA3FB6973A7BD76AA40B97D661D1992261457F7292F
- PowerShell-7.4.0-win-fxdependentWinDesktop.zip
  - 62D7972064C68C68B21ECD514B00D34301634929ACB9AF58A3F623CEE79C5ACB
- PowerShell-7.4.0-win-x64.msi
  - A9C8C5EEFA63289332CD1E4D1B1A904D332082847AA7D5055D662EEDE7C67B13
- PowerShell-7.4.0-win-x64.zip
  - 62151DB1D98A8B56AEB249CC8A3CE17948F1C83B4062DAC8D0C4302DE71CBD75
- PowerShell-7.4.0-win-x86.msi
  - DB199E57322EB0B385D6B1DB9421E54DAD0383D2F5399A3D61FC50CB87890035
- PowerShell-7.4.0-win-x86.zip
  - BCCC24E1F1DFC5D837111F3894D0CBA9A14CF9891EBC6645C63A7A584E54BC0A

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.4.0'>v7.4.0 Release of PowerShell</a>.</em>

**Comment by mario-marijan** [comment]
Hello, 

using PSCustomObject type accelerator to create object does not work for me in 7.4.0
e.g. `$CustomObject = [PSCustomObject]@{a=1;c=3;b=2;d=4}`
$CustomObject does not return anything.

Thanks

**Reply by VKrishna04** [reply]
i wish a winget command was made to update instantly and sometimes windows update did not identify this so would love to know how to get to update is easily instead of always downloading .msi

**Reply by nikeedev** [reply]
@VKrishna04 Just run `winget update Powershell` and it will do it right away. There is of course a little delay of when the updated version is released to the winget registry, so you can notice that the new version is already available on GitHub while there is hours or a day delay before it appears on winget

**Reply by VKrishna04** [reply]
When I do that It says 
```
Multiple installed packages found matching input criteria. Please refine the input.
Name       Id
-------------------------------
PowerShell 9MZ1SNWT0N5D        
PowerShell Microsoft.PowerShell
```
Now to note I have windows powershell pre-installed and this is `pwsh` or `powershell 7` so I don't know which one to choose

**Reply by diecknet** [reply]
> When I do that It says 
> ```
> Multiple installed packages found matching input criteria. Please refine the input.
> Name       Id
> -------------------------------
> PowerShell 9MZ1SNWT0N5D        
> PowerShell Microsoft.PowerShell
> ```
> Now to note I have windows powershell pre-installed and this is `pwsh` or `powershell 7` so I don't know which one to choose

`Microsoft.PowerShell` is PowerShell 7

**Comment by MartinDelzant** [comment]
Hello,

I do not see the linux-alpine-x64.tar.gz in your release.
Did you stop publishing that version ?
Looking forward to your answer,

Thank you very much

**Reply by adityapatwardhan** [reply]
The new name for the package is powershell-7.4.0-linux-musl-x64.tar.gz

Thank you!

**Reply by MartinDelzant** [reply]
Thanks !

**Comment by MrJia1997** [comment]
Is the `.msixbundle` version no longer released?

**Reply by adityapatwardhan** [reply]
It's available now. Thanks for letting us know.

**Comment by hegyizs** [comment]
Start-Process does not work anymore. Process ExitCode does not populated.

**Reply by chrispy81** [reply]
This bug should be high priority:
https://github.com/PowerShell/PowerShell/issues/20400

**Comment by ShreyasJejurkar** [comment]
No `.msixbundle` ?? @adityapatwardhan

**Reply by adityapatwardhan** [reply]
Let me check.

**Reply by adityapatwardhan** [reply]
It's available now. Thanks for reporting!

**Comment by wietsesas** [comment]
PSCustomObject objects cannot be created using `[PSCustomObject]@{x=1;y=2}`, this returns null. This makes this release completely unusable and dangerous to update on our production servers since all scripts will fail.

**Reply by diecknet** [reply]
Maybe related to issue #20719 ? (Constrained Language Mode)

**Comment by bl1te** [comment]
How do I update from 7.3.9? When I was installing it, I've checked "Update via Windows Update Center", but it does not show that 7.4.0 is available.
May I just launch the .msi file over the current installation? Or should I uninstall the previous version first?

**Reply by diecknet** [reply]
You can just manually install the newer `.msi`, no need to uninstall the previous version. 
It can take some time (2 weeks or so) until the new PS version is available with Windows Update.

**Comment by diecknet** [comment]
I saw that in the Download table the "Download (LTS)" column still links to PowerShell 7.2 and the "Download (stable)" links to PowerShell 7.4. 
Is that on purpose?
![grafik](https://github.com/PowerShell/PowerShell/assets/54552094/f6656376-f99f-4a28-86f2-e3014cbcaf56)

**Reply by JasonCraneAdkins** [reply]
Winget still offers the old 7.3.7.0

**Reply by nikeedev** [reply]
I think it’s correct. 7.2 is probably the LTS shouldn’t it be?

**Reply by diecknet** [reply]
> I think it’s correct. 7.2 is probably the LTS shouldn’t it be?

Yes, but 7.4 is the new LTS version, isn't it?

**Comment by paulomorgado** [comment]
I'm getting this from winget:

```
PS> winget list Microsoft.PowerShell                                                                                                                                                                                           
Name                   Id                   Version Available Source
--------------------------------------------------------------------
PowerShell 7.3.9.0-x64 Microsoft.PowerShell 7.3.9.0 7.4.0.0   winget

PS> winget upgrade Microsoft.PowerShell
A newer version was found, but the install technology is different from the current version installed. Please uninstall the package and install the newer version.
```

Any hints?

**Reply by diecknet** [reply]
There are multiple PowerShell install methods. Maybe you didnt install PowerShell initially using winget, but from the Microsoft Store instead?

**Reply by paulomorgado** [reply]
Yeah! That might be a winget issue. I just got the MSI mentioned in `winget show Microsoft.PowerShell` and installed it.

**Reply by MilioD** [reply]
I had the same issue, for me it's because Windows Update overwrites the PowerShell install technology with an .exe wrapper. Then you can no longer update it using WinGet. In my opinion this seems like an oversight in the update pipeline because the recommended install method is through WinGet.

Also for me the issue first showed up because the WinGet registry took longer to update the package than for it to validated in the Microsoft Update Catalog. So PowerShell auto updated via Windows Update before I could manually update it through the WinGet package manager.

**Reply by paulomorgado** [reply]
Or Windows Update should not take so long to update. :smile:

**Comment by B0R1K** [comment]
<img width="271" alt="image" src="https://github.com/PowerShell/PowerShell/assets/2473977/5786bf34-10ff-42d3-b452-aca5c30b1d42">
